### PR TITLE
fix: generation of `negative` flags

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
No need
**Summary**
fix generation of `negative` flags. currently, we have negative flags for all boolean flags regardless of `option.negative` value.

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Comment by alexander-akait** [comment]
@snitin315 we should better to solve it, I will finish your PR, dont' worry

**Comment by alexander-akait** [comment]
/cc @webpack/cli-team I want to clarify something here, we have three types:
- `type: 'boolean'` means `--option` and `--no-options`, any boolean for CLI always have positive and negative value
- `type: 'enum'` and `values: [true]` means only `--option` (no negative)
- `type: 'enum'` and `values: [false]` means only `--no-option` (no positive)

Here situation when you can use `type: 'enum'` and `values: [false, true]`, it is possible, but it doesn't make sense, because it means `boolean`

**Comment by alexander-akait** [comment]
Also need review, we should merge it before release

File: OPTIONS.md [hunk 1]
```diff
@@ -858,7 +858,7 @@ Commands:
   configtest|t [config-path]                                                         Validate a webpack configuration.
   help|h [command] [option]                                                          Display help for commands and options.
   info|i [options]                                                                   Outputs information about your system.
-  init|c [generation-path] [options]                                                 Initialize a new webpack project.
+  init|c [generation-path] [options]                                                 Initialize a new webpack configuration.
```
Comment (snitin315) [inline OPTIONS.md:861]: This should not have changed.

Comment (alexander-akait) [reply OPTIONS.md]: Good catch

**Comment by snitin315** [comment]
> Here situation when you can use `type: 'enum'` and `values: [false, true]`, it is possible, but it doesn't make sense, because it means `boolean`

Yes `'boolean'` should be preferred here.

**Review by anshumanv — DISMISSED** [review]
Much better

**Comment by alexander-akait** [comment]
@snitin315 Let's wait green https://github.com/webpack/webpack-cli/pull/2553 and merge, then rebase

**Review by alexander-akait — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Pull request merged**